guides/source/asset_pipeline.md

```diff
@@ -868,218 +906,350 @@ each asset will have a different file name when it is changed. This way you
+`RAILS_ASSET_URL` pseudo-method. Here’s an example:
 
-Assets Cache Store
-------------------
-
-By default, Sprockets caches assets in `tmp/cache/assets` in development
-and production environments. This can be changed as follows:
-
-```ruby
-config.assets.configure do |env|
-  env.cache = ActiveSupport::Cache.lookup_store(:memory_store,
-                                                { size: 32.megabytes })
-end
+```javascript
+export default class extends Controller {
+  init() {
+    this.img = RAILS_ASSET_URL("/icons/trash.svg");
+  }
+}
 ```
 
-To disable the assets cache store:
+This will transform into:
 
-```ruby
-config.assets.configure do |env|
-  env.cache = ActiveSupport::Cache.lookup_store(:null_store)
-end
+```javascript
+export default class extends Controller {
+  init() {
+    this.img = "/assets/icons/trash-54g9cbef.svg";
+  }
+}
 ```
 
-Adding Assets to Your Gems
---------------------------
+This ensures that the correct, digested file is used in your JavaScript code.
 
-Assets can also come from external sources in the form of gems.
+If you’re using bundlers like [Webpacker](https://webpack.js.org/) or
+[esbuild](https://esbuild.github.io/), you should let the bundlers handle the
+digesting process. If Propshaft detects that a file already has a digest in the
+filename (e.g., `script-2169cbef.js`), it will skip digesting the file again to
+avoid unnecessary reprocessing.
 
-A good example of this is the `jquery-rails` gem.
-This gem contains an engine class which inherits from `Rails::Engine`.
-By doing this, Rails is informed that the directory for this
-gem may contain assets and the `app/assets`, `lib/assets` and
-`vendor/assets` directories of this engine are added to the search path of
-Sprockets.
+For managing assets with [Import Maps](#importmap-rails), Propshaft ensures that
+assets referenced in the import map are appropriately handled and mapped to
+their digested paths during the precompilation process.
 
-Making Your Library or Gem a Pre-Processor
-------------------------------------------
+### Bypassing the Digest Step
 
-Sprockets uses Processors, Transformers, Compressors, and Exporters to extend
-Sprockets functionality. Have a look at
-[Extending Sprockets](https://github.com/rails/sprockets/blob/main/guides/extending_sprockets.md)
-to learn more. Here we registered a preprocessor to add a comment to the end
-of text/css (`.css`) files.
+If you need to reference files that refer to each other—like a JavaScript file
+and its source map—and want to avoid the digesting process, you can pre-digest
+these files manually. Propshaft recognizes files with the pattern
+`-[digest].digested.js` as files that have already been digested and will
+preserve their stable file names.
 
-```ruby
-module AddComment
-  def self.call(input)
-    { data: input[:data] + "/* Hello From my sprockets extension */" }
-  end
-end
-```
+### Excluding Directories from Digestion
```

Comment:
Yup! Thanks for calling this out. This and a few other sections were repeated because of my structure, however, I’ve re-ordered the sections so that the flow feels much more natural, and it helped me catch and remove some duplications.